localstack-core/localstack/services/lambda_/event_source_mapping/pollers/poller.py

```diff
@@ -52,7 +52,8 @@ def __init__(
         self.aws_region = parse_arn(source_arn)["region"]
         self.source_parameters = source_parameters or {}
         filters = self.source_parameters.get("FilterCriteria", {}).get("Filters", [])
-        self.filter_patterns = [json.loads(filter["Pattern"]) for filter in filters]
+        filter_patterns = [json.loads(event_filter["Pattern"]) for event_filter in filters]
+        self.filter_patterns = filter_patterns
```

Comment:
Why does this happen in two steps now?

Reply:
I had to debug here and having the evaluation at a different line makes it often easier, but it doesn't add anything here. I am happy to revert this change.

Reply:
reverted